tests/python_frontend/constant_and_keyword_args_test.py

```diff
@@ -556,6 +556,35 @@ def conditional_val(A: dace.float64[20], val: dace.compiletime):
     assert np.allclose(a, 0)
 
 
+def test_constant_propagation_with_normal_argument():
+
+    @dace.program
+    def conditional_val_with_access(A: dace.float64[20], val: dace.int32):
+        if val % 2 == 0:
+            A[:] = val
+        else:
+            A[:] = val + 1
+
+    a = np.random.rand(20)
+    sdfg = conditional_val_with_access.to_sdfg(a, 3, simplify=True, use_cache=True)
+    from dace.transformation.passes.constant_propagation import ConstantPropagation
+    res = dict()
+    ConstantPropagation().apply_pass(sdfg, res, {'val': 3})
+    sdfg.simplify()
+    assert sdfg.number_of_nodes() == 1
+    sdfg.view()
```

Comment:
Please remove viewing the SDFG from the test (unless it serves a specific test purpose?)